src/main/java/Luke.java

```diff
@@ -0,0 +1,246 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.*;
+
+public class Luke {
+
+    static String[] validCommands = {"bye", "list", "unmark", "mark", "todo", "event", "deadline", "delete"};
+
+    static String directoryPath = "./data";
+
+    static String fileName = "list.txt";
+
+    static File file = new File (directoryPath, fileName);
+
+    static boolean fileIsEmpty = true;
+
+    private static boolean isCommandValid(String command) {
+        for (String validCommand: validCommands) {
+            if (command.equals(validCommand)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public static void main(String[] args) {
+
+        String name = "Luke";
+
+        ArrayList<Task> taskList = new ArrayList<Task>();
+
+        ArrayList<String> saveTaskList = new ArrayList<String>();
+
+        int noTasks = 0;
+
+        try {
+            File directory = file.getParentFile();
+            if (!directory.exists()) {
+                directory.mkdirs();
+            }
+
+            if (!file.exists()) {
+                file.createNewFile();
+            }
+        } catch (IOException e) {
+            System.out.println("Failed to save file: " + e.getMessage());
+        }
+
+        try {
+            Scanner myReader = new Scanner(file);
+            while (myReader.hasNextLine()) {
+                String data = myReader.nextLine();
+                if (data != "") {
+                    saveTaskList.add(data);
+                    noTasks++;
+                }
+            }
+            myReader.close();
+            fileIsEmpty = saveTaskList.isEmpty();
+        } catch (FileNotFoundException e) {
+            System.out.println("File not found.");
+        }
+
+        saveTaskList.forEach(taskString -> {
+           String taskType = taskString.substring(1, 2);
+           switch (taskType) {
+               case "T":
+                   Todo todo = new Todo(taskString.substring(7).trim());
+                   taskList.add(todo);
+                   break;
+               case "D":
+                   String[] deadlineSplit = taskString.split("by: ");
+                   String deadlineDescription = deadlineSplit[0].substring(7, deadlineSplit[0].length() - 1);
+                   String by = deadlineSplit[1].substring(0, deadlineSplit[1].length() - 1).trim();
+                   Deadline deadline = new Deadline(deadlineDescription, by);
+                   taskList.add(deadline);
+                   break;
+               case "E":
+                   String[] eventFirstSplit = taskString.split("from: ");
```

Comment:
Maybe can describe what is the product after the first split and second split instead of naming them as eventFirstSplit and eventSecondSplit.